cl/people_db/filters.py

```diff
@@ -293,3 +365,31 @@ class Meta:
             "date_modified": DATETIME_LOOKUPS,
             "name": ALL_TEXT_LOOKUPS,
         }
+
+    def filter_roles(self, qs: QuerySet, name: str, value: bool) -> QuerySet:
+        """
+        Filters a QuerySet based on a many-to-many relationship involving the
+        `Role` model.
+
+        Args:
+            qs: The original QuerySet to be filtered.
+            name: The name of the many-to-many field to filter on.
+            value: The value to filter the many-to-many relationship with.
+
+        Returns:
+            The filtered QuerySet, prefetched with the filtered many-to-many
+            relationship.
+        """
+        if not value:
+            return qs
+
+        role_filters = self.get_filters_for_join_table(name)
+        if not role_filters:
+            return qs
+
+        prefetch = Prefetch(
+            name,
+            queryset=Role.objects.filter(**role_filters),
+            to_attr=f"filtered_{name}",
```

Comment:
Could there be a case where `to_attr` is different from `filtered_roles`? If so, perhaps make it fixed without using `{name}`. Otherwise, adjust accordingly in `get_parties_represented`.

Reply:
> Could there be a case where to_attr is different from filtered_roles?

@albertisfu We can use a different name. I simply used `filtered_roles` because we're using the `Role` model to construct the prefetch query. Would `filtered_parties_represented` be a more suitable alternative?

Reply:
Yeah, `filtered_roles` is fine, or `filtered_parties_represented`. I just wondered if it could change dynamically according to `name`, since it's set as `filtered_{name}`. However, it seems it will always be `filtered_roles`, so maybe there's no need to use `name` in it?

Reply:
Got it! I'll use a static name. Thanks!

Reply:
Great! :shipit: